plugins/flatpak/gs-flatpak-transaction.c

```diff
@@ -265,6 +302,25 @@ _transaction_operation_done (FlatpakTransaction *transaction,
 	}
 	switch (flatpak_transaction_operation_get_operation_type (operation)) {
 	case FLATPAK_TRANSACTION_OPERATION_INSTALL:
+		/* Handle special snowflake where "should-download" related refs for an installed ref
+		 * goes missing. In that case, libflatpak marks the main app ref as updatable
+		 * and then FlatpakTransaction resolves one of it's ops to install the related ref(s).
+		 *
+		 * We can depend on libflatpak till here. Since, libflatpak returns the main app
+		 * ref as updatable (instead of the related ref), we need to sync the main app's
+		 * statefor UI/UX.
+		 *
+		 * Map the current op's ref (which is related ref) to it's main app ref(which is currently
+		 * shown in the UI) and set the state of the main GsApp object back to INSTALLED here.
+		 *
+		 * This detection whether a related ref belongs to a main ref is quite sub-optimal as
+		 * of now.
+		 */
+		main_app = check_if_related_ref_is_installed (transaction, operation);
```

Comment:
Not sure if the question is valid, but what if for some reason the related ref gets installed before the main app? Or is the order guaranteed? If so, do we need to depend on some specific flatpak version for that to work?

Reply:
This is invalid as far as these changes are concerned. We expect the main-app/ref to be installed already, otherwise, how would we know if the app is updatable at the first place :)

Regarding the ordering, I think we should  just let FlatpakTransaction do it's magic